src/main/java/com/bookappstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderResponseDto> findAllOrders(Long userId, Pageable pageable) {
+        return orderMapper.toDtoList(orderRepository
+                .findOrdersByUserId(userId));
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto create(Long userId, OrderRequestDto orderRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCart(userId);
+        Order userOrder = orderRepository.save(
+                initializeNewOrder(shoppingCart, orderRequestDto));
+        orderItemRepository.saveAll(userOrder.getOrderItems()
+                .stream()
+                .peek(orderItem -> orderItem.setOrder(userOrder))
+                .collect(Collectors.toSet()));
+        return orderMapper.toDto(userOrder);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto setUpdateStatus(Long orderId, OrderUpdateDto orderUpdateDto) {
+        Order order = getOrderById(orderId);
+        Order.Status status = Order.Status.valueOf(orderUpdateDto
+                .status()
+                .toString()
+                .toUpperCase());
+        order.setStatus(status);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getItems(Long orderId, Long id) {
+        Order order = orderRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Can not find order by id:" + id));
+
+        return orderItemMapper.toDtos(order.getOrderItems());
+    }
+
+    @Override
+    public Set<OrderItemResponseDto> findAllOrderItems(Long orderId) {
+        Order order = orderRepository
+                .findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find order by id: " + orderId)
+                );
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    private Order initializeNewOrder(ShoppingCart shoppingCart, OrderRequestDto requestDto) {
+        Order newOrder = shoppingCartMapper.toOrder(shoppingCart);
+        newOrder.setShippingAddress(requestDto.shippingAddress());
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setStatus(Order.Status.NEW);
```

Comment:
When User place an order, which status should be? 
In entity you have Pending, here you have New. Figure out please

Reply:
@Modifying
    @Query("UPDATE Order o SET o.status = :status WHERE o.id = :orderId")
    void updateOrderByStatus(Long orderId, Order.Status status);
you said I should remove this method and query, but now it's not fixed, could you explain more accurate were I was wrong